src/System.IO/src/System/IO/TextWriter.cs

```diff
@@ -391,11 +391,43 @@ public virtual void WriteLine(decimal value)
         //
         public virtual void WriteLine(string value)
         {
-            if (value != null)
+            if (value==null)
             {
-                Write(value);
+                WriteLine();
             }
-            Write(CoreNewLineStr);
+            else
+            {
+                // We'd ideally like WriteLine to be atomic, in that one call
```

Comment:
Maybe; or maybe it would be best for the WriteLine special handling to be just in StreamWriter (that is used by console) because of it does not make sense in other places. Thoughs?

cc @stephentoub